src/python/pants/engine/legacy/query.py

```diff
@@ -133,7 +268,22 @@ def rules():
     owner_of_request,
     changes_since_request,
     changes_for_diffspec_request,
-    UnionRule(StaticTargetSet, OwnerOf),
-    UnionRule(StaticTargetSet, ChangesSince),
-    UnionRule(StaticTargetSet, ChangesForDiffspec),
+    # We get all these for free from the previous ability to convert to a BuildFileAddresses.
+    addresses_intersection_operator,
+    UnionRule(OperatorRequest, OwnerOf),
+    # TODO: better error message when paths can't be found through union rules!!! (just says no rule
+    # could be found because we didn't make the paths yet for these union members!)
+    UnionRule(OperatorRequest, ChangesSince),
+    UnionRule(OperatorRequest, ChangesForDiffspec),
+    UnionRule(OperatorRequest, Noop),
+    hydrate_noop,
+    noop_results,
+    # ???/intersection
+    RootRule(IntersectionOperands),
```

Comment:
Is this a root for testing purposes? If so, could add the root in the test instead...